src/Text.js

```diff
@@ -57,20 +57,18 @@ const computeTextStyle = (theme = defaultTheme, {
     }
   }
 
-  if (fontSize) {
```

Comment:
Now it's clear why index 0 was not working. However i think that this condition was to prevent som flow errors. If i remember correctly i think that you will get flow error that fontsize is undefined. Please check it. If so, setting fontsize as default prop should solve it ;-)